README.md

```diff
@@ -1,145 +1,308 @@
+indicated by `-dir` is mounted.
+
+**Example**
+
+Using `-fuse`, you do not need to specify instance names ahead of time:
+
+```
+./cloud_sql_proxy -dir=/cloudsql -fuse &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+* `-instances_metadata=metadata_key`
+
+Usable on [GCE](https://cloud.google.com/compute/docs/quickstart) only. The
+given [GCE metadata](https://cloud.google.com/compute/docs/metadata) key will be
+polled for a list of instances to open in `-dir`. The metadata key is relative
+from `computeMetadata/v1/`. The format for the value is the same as the
+'instances' flag. A hanging-poll strategy is used, meaning that changes to the
+metadata value will be reflected in the `-dir` even while the proxy is running.
+When an instance is removed from the list the corresponding socket will be
+removed from `-dir` as well (unless it was also specified in `-instances`), but
+any existing connections to this instance will NOT be terminated.
+
+**Example**
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances_metadata instance/attributes/<custom-metadata-key> &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
 
 Note: `-instances` and `-instances_metadata` may be used at the same time but
 are not compatible with the `-fuse` flag.
 
-cloud_sql_proxy authentication can be configured in a few different ways. Those listed higher on the list will override options lower on the list:
+* `-max_connections`
+
+If provided, the maximum number of connections to establish before refusing new
+connections. Defaults to 0 (no limit). Note: requests made through the
+SQL Admin API are rate limited to 2x the number of connections per minute.
+
+### Credentials and Authentication Flags
+
+Authentication can be configured in a few different ways. Those listed higher on
+the list will override options lower on the list:
 
 1. `credential_file` flag
 2. `token` flag
-3. Service account key at path stored in `GOOGLE_APPLICATION_CREDENTIALS`
-4. gcloud _user_ credentials (set from `gcloud auth login`)
-5. Default Application Credentials via goauth:
-
-   1. `GOOGLE_APPLICATION_CREDENTIALS` (again)
-   2. gcloud _application default_ credentials (set from ` gcloud auth application-default login`)
-   3. appengine.AccessToken (for App Engine Go < =1.9)
-   4. GCE/GAE metadata credentials
-
-When the proxy authenticates under the default service account of the
-Compute Engine VM it is running on the VM must have at least the
-sqlservice.admin API scope ("https://www.googleapis.com/auth/sqlservice.admin")
-and the associated project must have the SQL Admin API
-enabled.  The default service account must also have at least WRITER/EDITOR
-priviledges to any projects of target SQL instances.
-
-Specifying the `-credential_file` flag allows use of the proxy outside of
-Google's cloud. Simply [create a new service
+3. [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
```

Comment:
It's very important we don't loose #3 and #4 here - `gcloud auth login` is actually an entirely different than the `auth application-default` version, so both of these steps need to stay.

Reply:
Restored.